shell/edit/provisioning.cattle.io.cluster/Registries.vue

```diff
@@ -0,0 +1,149 @@
+<script>
+import { LabeledInput } from '@components/Form/LabeledInput';
+import { Banner } from '@components/Banner';
+import { Checkbox } from '@components/Form/Checkbox';
+import SelectOrCreateAuthSecret from '@shell/components/form/SelectOrCreateAuthSecret';
+import AdvancedSection from '@shell/components/AdvancedSection.vue';
+
+import RegistryConfigs from './RegistryConfigs';
+
+import RegistryMirrors from './RegistryMirrors';
+
+export default {
+  components: {
+    LabeledInput,
+    Banner,
+    Checkbox,
+    SelectOrCreateAuthSecret,
+    AdvancedSection,
+    RegistryConfigs,
+    RegistryMirrors
+  },
+
+  props: {
+    registerBeforeHook: {
+      type:     Function,
+      required: true,
+    },
+
+    mode: {
+      type:     String,
+      required: true,
+    },
+
+    value: {
+      type:     Object,
+      required: true,
+    },
+
+    showCustomRegistryInput: {
+      type:     Boolean,
+      required: true,
+    },
+
+    registryHost: {
+      type:     String,
+      required: false,
+      default:  null
+    },
+
+    registrySecret: {
+      type:     String,
+      required: false,
+      default:  null
+    },
+
+    showCustomRegistryAdvancedInput: {
+      type:     Boolean,
+      required: true,
+    },
+
+    isK3s: {
+      type:     Boolean,
+      required: true
+    },
+  }
+};
+</script>
+
+<template>
+  <div>
+    <div class="row">
+      <h3>{{ t('cluster.privateRegistry.label') }}</h3>
+    </div>
+    <div class="row">
+      <div class="col span-12">
+        <Banner
+          :closable="false"
+          class="cluster-tools-tip"
+          color="info"
+          label-key="cluster.privateRegistry.description"
+        />
+      </div>
+    </div>
+    <div class="row">
+      <Checkbox
+        :value="showCustomRegistryInput"
+        class="mb-20"
+        :label="t('cluster.privateRegistry.label')"
+        @input="$emit('custom-registry-changed', $event)"
+      />
+    </div>
+    <div
+      v-if="showCustomRegistryInput"
+      class="row"
+    >
+      <div class="col span-6">
+        <LabeledInput
+          v-model="registryHost"
```

Comment:
When `registryHost` is changed here that isn't propagated up to rke2.vue. This causes issues where it's referenced in `setRegistryConfig`